addons/pos_loyalty/tests/test_frontend.py

```diff
@@ -814,3 +814,63 @@ def test_gift_card_value_with_discount(self):
         # Check the created gift cards.
         self.assertEqual(len(programs['gift_card_1'].coupon_ids), 1)
         self.assertEqual(programs['gift_card_1'].coupon_ids.points, 50)
+
+    def test_promotion_rewards_per_order(self):
+        """
+        - Create a promotion with a reward of type "per_order",that applies on specific products
+        - Open the Pos and add the products to the order, and check that reward is added.
+        - Also check that there is no more reward applicable.
+        """
+        LoyaltyProgram = self.env['loyalty.program']
+        # Deactivate all other programs to avoid interference
+        (LoyaltyProgram.search([])).write({'pos_ok': False})
+
+        self.product_a = self.env["product.product"].create({
+            "name": "Test Product A",
+            "type": "product",
+            "list_price": 10,
+            "available_in_pos": True,
+            "taxes_id": [(6, 0, [])],
+        })
+        self.product_b = self.env["product.product"].create({
+            "name": "Test Product B",
+            "type": "product",
+            "list_price": 10,
+            "available_in_pos": True,
+            "taxes_id": [(6, 0, [])],
+        })
+        self.product_c = self.env["product.product"].create({
+            "name": "Test Product C",
+            "type": "product",
+            "list_price": 10,
+            "available_in_pos": True,
+            "taxes_id": [(6, 0, [])],
+        })
+
+        self.loyalty_program = self.env['loyalty.program'].create({
+            'name': 'Promo program - Per Order',
+            'program_type': 'promotion',
+            'pos_ok': True,
+            'rule_ids': [(0, 0, {
+                'reward_point_mode': 'order',
+                'reward_point_amount': 5,
```

Comment:
Yes, because the issue was that the reward is "10$ per order" so we should only be able to have 1 reward. Even if we have enough point to have 2 or more rewards this is a "per order" reward.